static/js/publisher/builds/components/searchSelect.js

```diff
@@ -0,0 +1,408 @@
+import React from "react";
+import PropTypes from "prop-types";
+
+class SearchSelect extends React.Component {
+  constructor(props) {
+    super(props);
+
+    // The available values should be the full list minus the current values
+    // sorted by name
+    const values = this.props.values
+      .filter(value => !this.props.value.includes(value.key))
+      .sort(this.sortByName);
+
+    this.state = {
+      // Get the correct objects for the currentValues as selected
+      selected: this.props.values.filter(value =>
+        this.props.value.includes(value.key)
+      ),
+      values: values,
+      searchTerm: "",
+      searchResults: values,
+      showSearch: false,
+      highlightedOption: null,
+      isMulti: this.props.isMulti || false,
```

Comment:
We should put default prop values in a `defaultProps` object. Like `SearchSelect.propTypes` you do:
```
SearchSelect.defaultProps = {
  isMulti: false,
  placeholder: ""
};
```